cmake/FetchCatch2.cmake

```diff
@@ -0,0 +1,46 @@
+# Use FetchContent to obtain Catch2.
+
+include(FetchContent)
+
+message(STATUS "Downloading Catch2...")
+
+function(fetch_catch2)
+    set(fetch_args "")
+    if ("${CMAKE_VERSION}" VERSION_GREATER_EQUAL "3.25.0")
+        set(fetch_args "SYSTEM")
+    endif()
+
+    FetchContent_Declare(
+        EP_Catch2
+
+        GIT_REPOSITORY https://github.com/catchorg/Catch2
+        GIT_TAG v3.7.0
+        GIT_SHALLOW TRUE
+        LOG_DOWNLOAD ON
+
+        ${fetch_args}
+
+        FETCHCONTENT_UPDATES_DISCONNECTED ON
+    )
+
+    FetchContent_GetProperties(EP_Catch2)
+
+    if(NOT ep_catch2_POPULATED)
+        # Avoid Catch2 compile warnings from being treated as errors.
+        string(REPLACE " -Werror" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+        string(REPLACE " -Werror" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
```

Comment:
Should `/WX` also be removed? (for MSVC)

Reply:
We do not pass `/WX` in any of our build or CI scripts, so it is not necessary.